src/state/__tests__/state-machine/expansion/hazardous-substance-regulation.spec.ts

```diff
@@ -0,0 +1,172 @@
+  const { send, getState, activatePolicy } = getTestActor({ useSpecialCards: true, playerCount: 1, seed });
+  const stateBefore = getState();
+
+  stateBefore.plantMarket.deck = concat(stateBefore.plantMarket.deck, stateBefore.plantMarket.table);
+  stateBefore.plantMarket.table = [];
+  stateBefore.animalMarket.deck = concat(stateBefore.animalMarket.deck, stateBefore.animalMarket.table);
+  stateBefore.animalMarket.table = [];
+
+  const plantCard = first(stateBefore.plantMarket.deck)!;
+  const animalCard = first(
+    filter(stateBefore.animalMarket.deck, (card) =>
+      plantCard?.habitats.every((habitat) => !card.habitats.includes(habitat)),
+    ),
+  )!;
+  stateBefore.players[0].hand.push(plantCard);
+  stateBefore.players[0].hand.push(animalCard);
+  stateBefore.plantMarket.deck = without(stateBefore.plantMarket.deck, plantCard);
+  stateBefore.animalMarket.deck = without(stateBefore.animalMarket.deck, animalCard);
+
+  activatePolicy({
+    policyName: "Hazardous substance regulation",
+    stateBefore,
+    specialCardSource: "plants",
+  });
+  send({ type: "user.click.player.hand.card", card: plantCard });
+  send({ type: "user.click.player.hand.card", card: animalCard });
+
+  const state = getState();
+  expect(state.commandBar).toBeDefined();
+  expect(
+    state.habitatMarket.deck.every((habitatTile) => habitatTile.name === "baltic" || !habitatTile.isAcquired),
+  ).toBe(true);
+});
+
+testRandomSeed("can only select plant cards in own hand", async (seed) => {
+  const { send, getState, activatePolicy } = getTestActor({ useSpecialCards: true, playerCount: 2, seed });
+  const stateBefore = getState();
+
+  stateBefore.plantMarket.deck = concat(stateBefore.plantMarket.deck, stateBefore.plantMarket.table);
+  stateBefore.plantMarket.table = [];
+  stateBefore.animalMarket.deck = concat(stateBefore.animalMarket.deck, stateBefore.animalMarket.table);
+  stateBefore.animalMarket.table = [];
+
+  const plantCards = filter(stateBefore.plantMarket.deck, (card) => card.habitats.includes("pelagic")).slice(0, 3)!;
+  const animalCards = filter(stateBefore.animalMarket.deck, (card) => card.habitats.includes("pelagic")).slice(0, 3)!;
+
+  stateBefore.plantMarket.deck = without(stateBefore.plantMarket.deck, ...plantCards);
+  stateBefore.animalMarket.deck = without(stateBefore.animalMarket.deck, ...animalCards);
+
+  stateBefore.players = stateBefore.players.map((player, index) => {
+    return {
+      ...player,
+      hand:
+        index === 0
+          ? [...player.hand.slice(0, 3), animalCards[index]]
+          : [...player.hand.slice(0, 2), plantCards[index], animalCards[index]],
+    };
+  });
+
+  stateBefore.plantMarket.table.push(plantCards[0]);
+  stateBefore.animalMarket.table.push(animalCards[0]);
+
+  activatePolicy({
+    policyName: "Hazardous substance regulation",
+    stateBefore,
+  });
+
+  plantCards.forEach((plantCard) => {
+    send({ type: "user.click.player.hand.card", card: plantCard });
```

Comment:
hmm.. plant card is currently in the plantMarket table  (line 108 above) why test is using `user.click.player.hand.card`

Reply:
The plant card is both in the player's hand and in the market. This is to make sure that only cards in the player's own hand can be played

Reply:
Ah. OK.

I think testing against unrealistic cases could lead to problem at some point. But if it's significantly easier than OK.

Reply:
Yeah you are right, I kinda took a shortcut here to not have to write two tests, but I agree I think it is better to split them up after all👍🏻